contracts/VRTVault/VRTVault.sol

```diff
@@ -0,0 +1,237 @@
+pragma solidity ^0.5.16;
+
+import "../Utils/SafeBEP20.sol";
+import "../Utils/IBEP20.sol";
+import "./VRTVaultProxy.sol";
+import "./VRTVaultStorage.sol";
+import "../Utils/WithAdmin.sol";
+import "../Utils/ReentrancyGuard.sol";
+
+contract VRTVault is VRTVaultStorage {
+    using SafeMath for uint256;
+    using SafeBEP20 for IBEP20;
+
+    /// @notice Event emitted when admin changed
+    event AdminTransfered(address indexed oldAdmin, address indexed newAdmin);
+
+    /// @notice Event emitted on VRT deposit
+    event Deposit(address indexed user, uint256 amount);
+
+    /// @notice Event emitted when accruedInterest and VRT PrincipalAmount is withrawn
+    event Withdraw(address indexed user, uint256 withdrawnAmount, uint256 totalPrincipalAmount, uint256 accruedInterest);
+
+    /// @notice Event emitted when Admin withdraw BEP20 token from contract
+    event WithdrawToken(address indexed tokenAddress, address indexed receiver, uint256 amount);
+
+    /// @notice Event emitted when accruedInterest is claimed
+    event Claim(address indexed user, uint256 interestAmount);
+
+    constructor() public {
+        admin = msg.sender;
+    }
```

Comment:
Why do we need the constructor at all here? To simplify testing?

Reply:
@kkirka for testing